### PR TITLE
Material theme for Dialog widget

### PR DESCRIPTION
**Pull request by jellison**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:**

Material theme for Dialog widget.

Close icon will (presumably) be fixed once the Material Icon theme has been ported to @widgets

Resolves dojo/themes#70

image

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-46hhjboea.now.sh
* Tests Deployment: https://dojowidgets-jcf3kkueh.now.sh

File: src/theme/material/dialog.m.css [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+	cursor: pointer;
+	font-size: 0;
+	padding: 8px;
+	position: absolute;
+	right: 16px;
+	top: 50%;
+	transform: translateY(-50%);
+}
+
+.close i {
```
Comment (matt-gadd) [inline src/theme/material/dialog.m.css]: I think this selector will cause a warning in the build

Comment (jellison) [reply src/theme/material/dialog.m.css]: You're right. I copied that from the Dojo theme, so I've fixed it as well.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-98v0yb7v3.now.sh
* Tests Deployment: https://dojowidgets-ns6fqcbat.now.sh

**Comment by tomdye** [comment]
@jellison i've noticed that when clicking on the dialog in material theme, it closes. Had something happened with the z-index of the underlay perhaps?

**Comment by jellison** [comment]
> @jellison i've noticed that when clicking on the dialog in material theme, it closes. Had something happened with the z-index of the underlay perhaps?

Good catch. This actually only effected the "non-visible" underlay variant. The way z-index is handled in the Material theme is incompatible with how the widget works. It's probably safer to handle the z-indexing ourselves anyway for consistency.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-24jb5bs92.now.sh
* Tests Deployment: https://dojowidgets-atux9rto6.now.sh

**Pull request merged**